File: spi-backend/src/main/java/dipl/spi/spi_backend/model/Account.java
package dipl.spi.spi_backend.model;

import dipl.spi.spi_backend.dto.AccountDto;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@NoArgsConstructor
@Getter
@Setter
public class Account {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String number;

    @Column(nullable = false)
    private String recipientName;

    @Column(nullable = false)
    private String recipientAddress;

    @ManyToOne(fetch = FetchType.LAZY)
    private AppUser creator;


    public void updateValues(AccountDto accountDto) {
        this.recipientName = accountDto.getRecipientName();
        this.recipientAddress = accountDto.getRecipientAddress();
    }

}

File: spi-frontend-mobile/src/app/dto/account.ts

export class Account {
    
    id: number;
    userId: number;
    
    number: string;
    recipientName: string;
    recipientAddress: string;

    constructor() {

    }

    updateValues(acc: Account) {
        this.id = acc.id;
        this.userId = acc.userId;
        this.number = acc.number;
        this.recipientName = acc.recipientName;
        this.recipientAddress = acc.recipientAddress;
    }

}